packages/transport/src/modules/sync/DatawalletModificationsProcessor.ts

```diff
@@ -117,57 +131,35 @@ export class DatawalletModificationsProcessor {
         this.cacheChanges.push(modification);
     }
 
-    private async applyUpdates() {
-        if (this.updates.length === 0) {
-            return;
-        }
-
-        for (const updateModification of this.updates) {
-            const targetCollection = await this.collectionProvider.getCollection(updateModification.collection);
-            const oldDoc = await targetCollection.read(updateModification.objectIdentifier.toString());
-
-            if (!oldDoc) {
-                throw new TransportError("Document to update was not found.");
-            }
-
-            const oldObject = Serializable.fromUnknown(oldDoc);
-            const newObject = { ...oldObject.toJSON(), ...updateModification.payload };
-
-            await targetCollection.update(oldDoc, newObject);
-        }
-    }
-
     private async applyCacheChanges() {
         if (this.cacheChanges.length === 0) {
             return;
         }
 
         this.ensureAllItemsAreCacheable();
 
-        const cacheChangesWithoutDeletes = this.cacheChanges.filter((c) => !this.deletes.some((d) => d.objectIdentifier.equals(c.objectIdentifier)));
+        const cacheChangesWithoutDeletes = this.cacheChanges.filter((c) => !this.deletedObjects.some((d) => c.objectIdentifier.equals(d)));
```

Comment:
This looks like a bug to me.

Reply:
But I didn't deleted the object Identifiers :D